File: typescript/easy/similarPairs.ts
function similarPairs(words: string[]): number {
  const map: Map<number, number> = new Map<number, number>();
  let rs = 0;
  for (const word of words) {
    let mask = 0;
    for (const c of word) {
      mask = mask | (1 << (c.charCodeAt(0) - "a".charCodeAt(0)));
    }
    if (!map.has(mask)) {
      map.set(mask, 0);
    }
    const pre = map.get(mask)!;
    rs += pre;
    map.set(mask, pre + 1);
  }
  return rs;
}

File: java/src/easy/SimilarPairs.java
package easy;

import java.util.Arrays;
import java.util.HashMap;

public class SimilarPairs {
    public int similarPairs(String[] words) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int rs=0;
        for(int i=0;i<words.length;i++){
            int m=0;
            for(int k=0;k<words[i].length();k++){
                m=m|(1<<(words[i].charAt(k)-'a'));
            }
            map.putIfAbsent(m,0);
            int pre=map.get(m);
            rs+=pre;
            map.put(m,pre+1);
        }
        return rs;
    }
}
